### PR TITLE
Compare Final Config for Testing

### PR DESCRIPTION
**Pull request by FSVetaz**
Print out final eslint config with the rules sorted alphabetically. Then it will be compared to the previous reason.

## To-Dos
- [x] Run tests (part of pre-push hook)
- [x] Update demo and tests, if linting configuration is being changed
- [ ] Update documentation & README
- [ ] Increment package.json version

### CODE REVIEW

**Comment by skye2k2** [comment]
Making these PRs simultaneously results in abundant merge conflicts.

The purpose of having the output saved in a readable (multiline) format is so that the _humans_ can see what changed. Shoving everything into one line might make the test output cleaner, but makes the git diff a nightmare.

**Comment by skye2k2** [comment]
`npm test` now creates an untracked `demo/test/snapshots/new-lint-results.txt-e` file with the original content.

Tests almost pass, but `new-lint-final-config.json` has a developer-specific entry:

`"parser": "/home/vetaz/fs-webdev/eslint-config-tree/node_modules/babel-eslint/lib/index.js"`

**Comment by FSVetaz** [comment]
> `npm test` now creates an untracked `demo/test/snapshots/new-lint-results.txt-e` file with the original content.
> 
> Tests almost pass, but `new-lint-final-config.json` has a developer-specific entry:
> 
> `"parser": "/home/vetaz/fs-webdev/eslint-config-tree/node_modules/babel-eslint/lib/index.js"`

I think you need to do an `npm i` and then rerun the tests. Also I removed the .json file, that one shouldn't be committed.

**Comment by FSVetaz** [comment]
> > `npm test` now creates an untracked `demo/test/snapshots/new-lint-results.txt-e` file with the original content.
> > Tests almost pass, but `new-lint-final-config.json` has a developer-specific entry:
> > `"parser": "/home/vetaz/fs-webdev/eslint-config-tree/node_modules/babel-eslint/lib/index.js"`
> 
> I think you need to do an `npm i` and then rerun the tests. Also I removed the .json file, that one shouldn't be committed.

Oh yeah I should remove the developer specific thing

**Comment by skye2k2** [comment]
I did not like the `-e` extension added to the output file. Can you test to see if using a `.tmp` extension breaks in your version of `sed`?

**Comment by FSVetaz** [comment]
> I did not like the `-e` extension added to the output file. Can you test to see if using a `.tmp` extension breaks in your version of `sed`?

That doesn't work unfortunately.
image

**Comment by skye2k2** [comment]
And if it is quoted `'.tmp'`?

**Comment by FSVetaz** [comment]
> And if it is quoted `'.tmp'`?

nope. You use FreeBSD? instead of OS X's sed?

File: demo/example.js [hunk 1]
```diff
@@ -12,6 +12,10 @@
 // Hack: Note that these work, regardless of case
 // Here be Dragons
 
+/**
```
Comment (FSVetaz) [inline demo/example.js:15]: Ohh that is interesting

**Review by skye2k2 — APPROVED** [review]
Thanks for your diligent work on this. I feel much more secure in making new changes, now.

**Pull request merged**